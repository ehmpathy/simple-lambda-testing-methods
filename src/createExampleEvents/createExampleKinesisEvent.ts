import { KinesisStreamEvent } from 'aws-lambda';
import { EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED } from './eventMetadataWhichShouldNotBeNeeded';

/**
 * creates a kinesis event for each message you give it:
 * - matches the shape of the kinesis event that aws would have triggered your lambda with
 * - automatically base64 encodes your message just like kinesis will do
 */
export const createExampleKinesisEvent = ({ messages }: { messages: string[] }): KinesisStreamEvent => ({
  Records: messages.map((message) => {
    return {
      awsRegion: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventID: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventName: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventSource: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventSourceARN: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventVersion: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      invokeIdentityArn: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      kinesis: {
        approximateArrivalTimestamp: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        data: Buffer.from(message).toString('base64'), // base64 encode the message, like kinesis will do
        kinesisSchemaVersion: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        partitionKey: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        sequenceNumber: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      },
    };
  }),
});
