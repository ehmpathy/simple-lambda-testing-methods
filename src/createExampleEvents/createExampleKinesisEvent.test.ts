import { KinesisStreamEvent } from 'aws-lambda';
import { createExampleKinesisEvent } from './createExampleKinesisEvent';

describe('createExampleKinesisEvent', () => {
  it('should create an accurate looking sns event', () => {
    const event: KinesisStreamEvent = createExampleKinesisEvent({ messages: ['hello', 'there'] });
    expect(event).toMatchSnapshot();
  });
});
