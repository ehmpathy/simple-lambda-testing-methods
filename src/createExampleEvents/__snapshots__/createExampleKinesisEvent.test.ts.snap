// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createExampleKinesisEvent should create an accurate looking sns event 1`] = `
Object {
  "Records": Array [
    Object {
      "awsRegion": undefined,
      "eventID": undefined,
      "eventName": undefined,
      "eventSource": undefined,
      "eventSourceARN": undefined,
      "eventVersion": undefined,
      "invokeIdentityArn": undefined,
      "kinesis": Object {
        "approximateArrivalTimestamp": undefined,
        "data": "aGVsbG8=",
        "kinesisSchemaVersion": undefined,
        "partitionKey": undefined,
        "sequenceNumber": undefined,
      },
    },
    Object {
      "awsRegion": undefined,
      "eventID": undefined,
      "eventName": undefined,
      "eventSource": undefined,
      "eventSourceARN": undefined,
      "eventVersion": undefined,
      "invokeIdentityArn": undefined,
      "kinesis": Object {
        "approximateArrivalTimestamp": undefined,
        "data": "dGhlcmU=",
        "kinesisSchemaVersion": undefined,
        "partitionKey": undefined,
        "sequenceNumber": undefined,
      },
    },
  ],
}
`;
